### ISSUE TITLE
[Bug]: Function Multi Versioning. Fix crashes, adjust names and mangling, support Android API 23-29 and rcpc3 feature.

### ISSUE THREAD

**Issue by ilinpv** [android/ndk#1947]
### Description

LLVM commits to cherry-pick
https://github.com/llvm/llvm-project/commit/30b0fdfff1931c625babe45cbf9405003c48f8a0
https://github.com/llvm/llvm-project/commit/8ec50d64464c7e6804827d5f4f298366be535001
https://github.com/llvm/llvm-project/commit/b04fe222d21277f8359130cd50ae7303b3d15254
https://github.com/llvm/llvm-project/commit/e5fe3d27fc1d5f27fe107f3380f3727362bac66c

### Commit to cherry-pick

30b0fdfff1931c625babe45cbf9405003c48f8a0, 8ec50d64464c7e6804827d5f4f298366be535001, b04fe222d21277f8359130cd50ae7303b3d15254, e5fe3d27fc1d5f27fe107f3380f3727362bac66c

### Affected versions

r26

### Host OS

Linux, Windows

### Host OS version

Ubuntu 22.04 LTS

### Affected ABIs

arm64-v8a

**Comment by ilinpv** [comment]
Preparing patch of cherry-picks to llvm-r487747 for next NDK respin.

**Comment by DanAlbert** [comment]
These aren't regressions, are they? FMV wasn't in r25 at all, I thought.

**Comment by pirama-arumuga-nainar** [comment]
IIUC, it's a new feature in `r26` toolchain that's broken at launch.  One exception is that https://github.com/llvm/llvm-project/commit/b04fe222d21277f8359130cd50ae7303b3d15254 seems like a feature rather than a bug fix to me.

**Comment by DanAlbert** [comment]
I'm inclined to say that we should not be backporting changes for experimental clang features that were not ready yet when we branched the toolchain. If these had been filed before we reached the release candidate that'd probably be a different story, but it's hard to justify risking the stability of the release with fixes for unfinished features. We did that last year with r25 and aiui that was the cause of https://github.com/android/ndk/issues/1862. It's critical that point releases do not introduce regresssions, so unless the cherry-pick is _fixing_ a regression from the previous major release, it's not typically a candidate.
